### ISSUE TITLE
Implement numpy.max for language C

### ISSUE THREAD

**Issue by spossann** [pyccel/pyccel#1621]
The `.max()` method seems to work in Fortran, but not in C:

```python
# test2.py

from numpy import arange
from pyccel.epyccel import epyccel

def np_max(a: 'float[:]') -> float:
    return a.max()

a = arange(4)*0.1
print(f'{np_max(a) = }')

np_max_fortran = epyccel(np_max)
print(f'{np_max_fortran(a) = }')

np_max_c = epyccel(np_max, language='c')
print(f'{np_max_c(a) = }')
```
which at the moment (pyccel 1.10.0) yields
```#python
19:57 $ python test2.py
np_max(a) = 0.30000000000000004
np_max_fortran(a) = 0.30000000000000004

ERROR at code generation stage
pyccel:
 |fatal [codegen]: mod_3dvgqxipuquy_3dvgqxipuquy.py| Pyccel has encountered syntax that has not been implemented yet. Please create an issue at https://github.com/pyccel/pyccel/issues and provide a small example of your problem. Do not forget to specify your target language.

```

**Comment by EmilyBourne** [comment]
Indeed, we still have not agreed on the best approach for functions such as this. We have an internal stdlib here:
https://github.com/pyccel/pyccel/tree/devel/pyccel/stdlib
Where we implement functions which are missing from the target language,  however it is not clear whether we should rewrite lots of functions in different target languages or whether we should have a Python implementation which we can pyccelise to the target language.

In this case (at least for now) I think it should be sufficient to write a function in :
https://github.com/pyccel/pyccel/tree/devel/pyccel/stdlib/ndarrays
And add the necessary function in the codegen stage to call this function

**Comment by abastola0** [comment]
@spossann  can you please assign me on this issue?

**Comment by spossann** [comment]
@abastola0 I think you have to ask @EmilyBourne, I am not a maintainer

**Comment by EmilyBourne** [comment]
@abastola0 Do you have enough information to tackle this issue? Feel free to contact me on [discord](https://discord.gg/2Q6hwjfFVb) if you have any trouble getting your environment set up/understanding the docs/etc

**Comment by EmilyBourne** [comment]
Fixed by #1640
